lib/es7.js

```diff
@@ -13,8 +13,8 @@ async function asyncPool(poolLimit, array, iteratorFn) {
   }
   const ret = [];
   const executing = [];
-  for (const item of array) {
-    const p = Promise.resolve().then(() => iteratorFn(item, array));
+  for (const [idx, item] of array.entries()) {
+    const p = Promise.resolve().then(() => iteratorFn(item, idx, array));
```

Comment:
Let's use the full word?

```suggestion
  for (const [index, item] of array.entries()) {
    const p = Promise.resolve().then(() => iteratorFn(item, index, array));
```